Initial commit of Docstrings

wojciechwisniewski opened the pull request:
Docstrings added for:
- api.py/add()
- api.py/clear()
- api.py/close()
- api.py/flush()
-__init__.py/remove()
-__init__.py/rename()

wojciechwisniewski committed:
Initial commit of Docstrings

Docstrings added for:
- api.py/add()
- api.py/clear()
- api.py/close()
- api.py/flush()
-__init__.py/remove()
-__init__.py/rename()

premopie commented on __init__.py:
```diff
@@ -242,7 +242,10 @@ def __repr__(self):
 
-    '''Each instance of the Variable class is paired with an associated
+    '''
+    Notes
```
Why Notes?

premopie commented on __init__.py:
```diff
@@ -251,14 +254,44 @@ def match(self):
-
-        From HDF5 file remove group paired with this variable.
-
+        Removes associated HDF5 group.
```
I would keep the first sentence "From HDF5 file remove group paired with this variable."

premopie commented on __init__.py:
```diff
@@ -251,14 +254,44 @@ def match(self):
+        
+        Examples
+        ---------
+        Add example variable (named "E") to the HDF5 file with index = 0 in the workspace::
```
Please change naming convention for variables. var1, var2, ... seems to be fine.
Also default index = -1 would be better.

premopie commented on __init__.py:
```diff
@@ -251,14 +254,44 @@ def match(self):
+        ---------
+        Add example variable (named "E") to the HDF5 file with index = 0 in the workspace::
+    
+        >>> <workspace_name>[0].create.basic("E")
```
Actually, the object <workspace_name> is not a name of the workspace :)

premopie commented on api.py:
```diff
@@ -6,6 +6,40 @@
 
 def add(filename, mode='a'):
+    """
+    Creates a HDF5 file in the workspace.
```
Files are created on disk rather than in the workspace!

premopie commented on api.py:
```diff
@@ -6,6 +6,40 @@
+    ------
+    Based on **h5py.File()**.
+    
+    Every HDF5 file is also a HDF5 group representing the root group of the file.
```
???

premopie commented on api.py:
```diff
@@ -23,6 +57,21 @@ def add(filename, mode='a'):
 
 def clear():
+    """
+    Removes all HDF5 files from the workspace.
```
Sounds scary. We should discuss terminology.

premopie commented on api.py:
```diff
@@ -31,15 +80,28 @@ def clear():
-    dowiązania do obiektów z usuwanego zasobu i zamknij plik.
-
+    """close()
+    Closes selected HDF5 file and removes it from the workspace.
```
Same as above.

wojciechwisniewski committed:
Documentation updated

- docstrings for SPHINX documentation updated;
- index.rst for SPHINX documentation placed in docs/directory
- README.md extended to contain detailed project description;
